pwiz_tools/Skyline/Controls/Graphs/GraphSpectrum.cs

```diff
@@ -911,6 +987,240 @@ private void ClearGraphPane()
         }
 
         public void UpdateUI(bool selectionChanged = true)
+        {
+            _updateManager.QueueUpdate(false);
+        }
+
+        private class UpdateManager : IDisposable
+        {
+            private readonly GraphSpectrum _parent;
+            private readonly Timer _timer;
+
+            private Cache _spectraCache;
+            private Cache _spectraLoadCache;
+            private Cache _precursorStringCache;
+
+            public UpdateManager(GraphSpectrum parent)
+            {
+                _parent = parent;
+                _timer = new Timer { Interval = 100 };
+                _timer.Tick += DoUpdate;
+            }
+
+            public bool IsUpdating
+            {
+                get
+                {
+                    lock (_timer)
+                    {
+                        return _timer.Tag != null;
+                    }
+                }
+            }
+
+            public void QueueUpdate(bool isUserAction)
+            {
+                // Restart the timer at 100ms, giving the UI time to interrupt.
+                lock (_timer)
```

Comment:
It looks to me like everything involving this timer is happening on the UI thread. Is anything here happening on a background thread that would make this "lock" necessary?

Reply:
GraphSpectrum implements IUpdatable, I think it's safer to use a lock since UpdateUI could potentially be called from anywhere?